src/optimization.py

```diff
@@ -54,6 +54,12 @@ def optimize_hyperparameter(X_train, y_train, pipe,
     >>> results = optimize_hyperparameter(X_train, y_train, pipe, param_grid)
     
     """
+    if (len(X_train) == 0 | len(y_train) == 0):
```

Comment:
Super minor, but usually you wouldn't use bitwise operator `|`, but there's no real difference here other than `|` is not short-circuiting while `or` is. ( `|` always evaluates both sides of the expression even if the first part is true, while `or` knows it doesn't need to check the second part if the first part already evaluated to True). 

You can also just check `if not X_train or not y_train` since an empty list `[]` evaluates to False